tests/VObject/ITip/BrokerTester.php

```diff
@@ -45,9 +45,7 @@ public function process($input, $existingObject = null, $expected = false)
 
         $vcal = Reader::read($input);
 
-        foreach ($vcal->getComponents() as $mainComponent) {
-            break;
-        }
+        $mainComponent = $vcal->getComponents()[0];
```

Comment:
are we sure `$vcal->getComponents()` is never empty at this point?

Reply:
No, we are not. But neither were we before. This only makes sure that the variable is set. If `getComponents` is empty, an error would be thrown.

Reply:
I guess with this change you traded one php warnings with another :-).
I think phpstan will yell at us for this new line on a higher level.

before your change we had a "maybe undefined variable".
with this change we might get a "undefined offset on array" warning

Reply:
That's correct.

I struggled with this, as the alternative would be to set `$mainComponent` to null before the loop. But then, we'd potentially call `->name` on null, which we can't do either.

There are many potential fixes for this problem: add an assert, add an if, and then some. But as this is a test, I'd like to keep focus on readability and not change too much. If phpstan will yell at us later, we'll fix it later. We might not even make it to the level at which this becomes a problem.

Reply:
ohh I didn't realize we are in a test here. leave it as is then. thx for the note.